src/endpoints/admin.rs

```diff
@@ -23,12 +23,12 @@ struct AdminTemplate<'a> {
     update: &'a Option<Version>,
 }
 
-pub async fn get_admin() -> Result<impl IntoResponse, AppError> {
+pub async fn get_admin(State(data): State<AppState>) -> Result<impl IntoResponse, AppError> {
```

Comment:
maybe: `State(AppState { args, .. }): ...`

Then you do not need `data.args` everywhere.